operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/backup/ElasticsearchBackupRepository.java

```diff
@@ -307,7 +313,66 @@ private List<SnapshotInfo> findSnapshots(String repositoryName, Long backupId) {
     }
   }
 
-  private GetBackupStateResponseDto getBackupResponse(Long backupId, List<SnapshotInfo> snapshots) {
+  protected boolean waitForFinishedSnapshotWithTimeout(
```

Comment:
I think the method name is confusing, because it return boolean, but we always wait, whether the output is true or false. I would rename this method to something like `isSnapshotFinishedWithinTimeout()`